Lib/Allocator.cpp

```diff
@@ -33,18 +33,27 @@ void Lib::setMemoryLimit(size_t limit) { LIMIT = limit; }
 
 // override global allocators to keep track of allocated memory, doing very little else
 // TODO does not support get_new_handler/set_new_handler as we don't use it, but we could
-void *operator new(size_t size) {
+void *operator new(size_t size, std::align_val_t align) {
   if(ALLOCATED + size > LIMIT)
     throw std::bad_alloc();
   ALLOCATED += size;
   {
     Lib::TimeoutProtector tp;
-    if(void *ptr = std::malloc(size))
+    if(void *ptr = std::aligned_alloc(static_cast<size_t>(align), size))
```

Comment:
TODO this needs to be fixed on Mac. `size` needs to be a multiple of `align`, apparently.

Reply:
(And it might not like being called with `align==1`.)